server/game/cards/03_TWI/units/SabineWrenYouCanCountOnMe.ts

```diff
@@ -0,0 +1,43 @@
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import AbilityHelper from '../../../AbilityHelper';
+import { AbilityRestriction, Aspect, RelativePlayer, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class SabineWrenYouCanCountOnMe extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '2843644198',
+            internalName: 'sabine-wren#you-can-count-on-me',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addConstantAbility({
+            title: 'While this unit is exhausted, she can\'t be attacked',
+            condition: (context) => context.source.exhausted,
+            ongoingEffect: AbilityHelper.ongoingEffects.cardCannot(AbilityRestriction.BeAttacked)
+        });
+
+        this.addOnAttackAbility({
+            title: 'Discard the top card from your deck',
+            optional: true,
+            immediateEffect: AbilityHelper.immediateEffects.discardSpecificCard((context) => ({
+                target: context.source.controller.getTopCardOfDeck()
+            })),
+            ifYouDo: (ifYouDoContext) => ({
+                title: 'If it doesn\'t share an aspect with your base, deal 2 damage to a ground unit',
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
+                    condition: (context) => !ifYouDoContext.events[0].card.aspects.includes(context.source.controller.base.aspects[0]),
```

Comment:
Could we change this so that it would work in case there was ever a base that had more than one aspect